### ISSUE TITLE
Network Events source on Mac OSX

### ISSUE THREAD

**Issue by mikewehr** [open-ephys/GUI#341]
Any advice on how to get the Network Events source to show up on a Mac OSX install of the master binary? I have installed zmq using macports, and also tried with zmq3, but Network source does not appear in the GUI. It appears all the right files are in /opt/local/lib, and matlab_zeroMQ_wrapper_example.m executes fine, which suggests that zmq is working ... what else can I check about the zmq installation?
Works fine for me on Ubuntu, was hoping for platform independence. 
-Mike

$ ls  /opt/local/lib
libsodium.18.dylib  libzmq.3.dylib      pkgconfig
libsodium.a     libzmq.a
libsodium.dylib     libzmq.dylib

**Comment by jonnew** [comment]
Are you using the plugin-GUI? I think this is a plugin that will need to be
compiled separately from the main GUI.

Jon

On Tue, Jul 5, 2016 at 8:57 PM, mikewehr notifications@github.com wrote:

> Any advice on how to get the Network Events source to show up on a Mac OSX
> install of the master binary? I have installed zmq using macports, and also
> tried with zmq3, but Network source does not appear in the GUI. It appears
> all the right files are in /opt/local/lib, and
> matlab_zeroMQ_wrapper_example.m executes fine, which suggests that zmq is
> working ... what else can I check about the zmq installation?
> Works fine for me on Ubuntu, was hoping for platform independence.
> -Mike
> 
> $ ls /opt/local/lib
> libsodium.18.dylib libzmq.3.dylib pkgconfig
> libsodium.a libzmq.a
> libsodium.dylib libzmq.dylib
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/open-ephys/GUI/issues/341, or mute the thread
> https://github.com/notifications/unsubscribe/ABXP45W-7xl0Mg63dn3GwnFarRKNQil8ks5qSv1lgaJpZM4JFqe2
> .

## 

Jonathan Newman
Postdoctoral Fellow, MIT
http://www.mit.edu/~jpnewman/

**Comment by mikewehr** [comment]
Thanks Jon. I don’t understand why it would need to be compiled separately if I’m using the downloaded binaries, and the Network Events source works fine with an ubuntu install. That’s why I’m suspicious of the zmq dependencies on Mac, but I’m at a loss how to troubleshoot those.
-Mike

**Comment by mikewehr** [comment]
Indeed, it works after compiling from source. Thanks Jon!

**Comment by jonnew** [comment]
@mikewehr: yeah, its always the best solution. Maintaining all those binaries is a full time job that I'm glad I have nothing to do with :) @jsiegle and @jvoigts You may want to reopen this since the pre-compiled binaries don't seem to come with the plugins which takes away about 80% of their functionality.

EDIT: I just realized this is on the GUI repo rather than plugin-GUI. So this might be another issue entirely or it might be exactly as stated if the pre-compiled stuff is created with plugin-GUI source

**Comment by mikewehr** [comment]
```
Can't people still find the thread if it's closed?


On 7/6/16 8:31 PM, Jon wrote:


  @mikewehr: yeah, its always the best
    solution. Maintaining all those binaries is a full time job that
    I'm glad I have nothing to do with :) @jsiegle
    and @jvoigts You may want to reopen this
    since the pre-compiled binaries don't seem to come with the
    plugins which takes away about 80% of their functionality.
  —
    You are receiving this because you were mentioned.
    Reply to this email directly, view
      it on GitHub, or mute
      the thread.
```

**Comment by jonnew** [comment]
1. Not really because it gets buried in the "closed" section which is not
   searched by default on github.
2. Understand that issues are not "bad" they are just a statement of fact.
   This is not peer review :).

On Wed, Jul 6, 2016 at 11:46 PM, mikewehr notifications@github.com wrote:

> Can't people still find the thread if it's closed?
> 
> On 7/6/16 8:31 PM, Jon wrote:
> 
> @mikewehr: yeah, its always the best
> solution. Maintaining all those binaries is a full time job that
> I'm glad I have nothing to do with :) @jsiegle
> and @jvoigts You may want to reopen this
> since the pre-compiled binaries don't seem to come with the
> plugins which takes away about 80% of their functionality.
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view
> it on GitHub, or mute
> the thread.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/open-ephys/GUI/issues/341#issuecomment-230971680, or mute
> the thread
> https://github.com/notifications/unsubscribe/ABXP43mf8IOlLBNScu66Iq9hM3MWgRDGks5qTHangaJpZM4JFqe2
> .

## 

Jonathan Newman
Postdoctoral Fellow, MIT
http://www.mit.edu/~jpnewman/

**Comment by mikewehr** [comment]
I'm re-opening this issue for awareness. The problem is solved by compiling from source code.

**Comment by jsiegle** [comment]
Hmmm...which OS X version are you using? It might be due to the fact that I compiled zmq for 10.11. I've been meaning to re-compile to make it backwards-compatible. Is there a chance this is the issue?

**Comment by mikewehr** [comment]
Well, recompiling solved the problem for me (El Capitan 10.11.5). So my best guess is that was the issue.
